kythe/java/com/google/devtools/kythe/extractors/java/JavaCompilationUnitExtractor.java

```diff
@@ -707,13 +694,15 @@ private void processRequiredInput(
       }
     }
 
-    // Identify generated sources by checking if the source file is under the genSrcDir.
-    if (genSrcDir.isPresent()
-        && !isJarPath
-        && requiredInput.getKind() == Kind.SOURCE
-        && Paths.get(relativePath).startsWith(genSrcDir.get())) {
+    // Identify generated sources by checking if the source file is under the SOURCE_OUTPUT
+    // directory.
+    if (isGeneratedSource(fileManager, requiredInput)) {
       results.explicitSources.add(strippedPath);
-      results.newSourcePath.add(genSrcDir.get().toString());
+      Iterables.addAll(
```

Comment:
This looks like we're adding all the generated sources every time we handle a required input instead of just once.

Reply:
`newSourcePath` is a set and this mirrors the old behavior.  We're adding all of the generated source directories to the `newSourcePath` set for each generated source we encounter.

Reply:
That shouldn't change between generated sources, though.  We should just create the set somewhere once to be more explicit.